src/app/containers/MostRead/index.jsx

```diff
@@ -1,28 +1,47 @@
 import React, { useContext } from 'react';
+import { string } from 'prop-types';
 import { RequestContext } from '#contexts/RequestContext';
 import { ServiceContext } from '#contexts/ServiceContext';
 import useToggle from '../Toggle/useToggle';
 import Canonical from './Canonical';
 
-const getLocalMostReadEndpoint = (service, variant) => {
-  const localhostURL = 'http://localhost:7080';
-  const localServiceURL = `${localhostURL}/${service}`;
+const getMostReadEndpoint = ({ service, variant }) =>
+  variant
+    ? `/${service}/most_read/${variant}.json`
+    : `/${service}/most_read.json`;
 
-  return variant
-    ? `${localServiceURL}/most_read/${variant}.json`
-    : `${localServiceURL}/most_read.json`;
-};
-
-const MostReadContainer = () => {
+const MostReadContainer = ({ endpointOverride }) => {
   const { variant } = useContext(RequestContext);
-  const { service } = useContext(ServiceContext);
+  const { service, script, mostRead, dir, datetimeLocale } = useContext(
+    ServiceContext,
+  );
 
   const { enabled } = useToggle('mostRead');
   if (!enabled) {
     return null;
   }
 
-  return <Canonical endpoint={getLocalMostReadEndpoint(service, variant)} />;
+  const endpoint =
+    endpointOverride || getMostReadEndpoint({ service, variant });
+
+  return (
+    <Canonical
+      endpoint={endpoint}
+      script={script}
+      service={service}
+      translations={mostRead}
+      dir={dir}
+      locale={datetimeLocale}
```

Comment:
To clean up this code, we can move https://github.com/bbc/simorgh/pull/5023/files#diff-d1eac320f7b67f0b8810c3fbf7f10dc3R15-R17 and https://github.com/bbc/simorgh/pull/5023/files#diff-d1eac320f7b67f0b8810c3fbf7f10dc3R30-R34 to `Canonical/index.jsx` so we don't have to pass them as props.